Bug: JSON validation error for chrome-devtools.json in validate-mcp-integrations.sh

## Description

Running the MCP validation script results in:


However, direct JSON validation passes:
{
    "claude_code_command": "claude mcp add-json chrome-devtools --scope user '{\"type\":\"stdio\",\"command\":\"npx\",\"args\":[\"chrome-devtools-mcp@latest\",\"--channel=canary\",\"--headless=true\",\"--isolated=true\",\"--viewport=1920x1080\",\"--logFile=/tmp/chrome-mcp.log\"]}'",
    "mcpServers": {
        "chrome-devtools": {
            "command": "npx",
            "args": [
                "chrome-devtools-mcp@latest",
                "--channel=canary",
                "--headless=true",
                "--isolated=true",
                "--viewport=1920x1080",
                "--logFile=/tmp/chrome-mcp.log"
            ]
        }
    }
}

## Steps to Reproduce

1. Run [0;35mMCP Integrations Validation[0m

[0;35mTesting Prerequisites[0m
[0;34m[INFO][0m Testing: Node.js availability
[0;32m[SUCCESS][0m Node.js availability: PASSED
[0;34m[INFO][0m Testing: npm availability
[0;32m[SUCCESS][0m npm availability: PASSED
[0;34m[INFO][0m Node.js version: v25.8.1
[0;34m[INFO][0m npm version: 11.11.0

[0;35mTesting Network Connectivity[0m
[0;34m[INFO][0m Testing: Internet connectivity
[0;32m[SUCCESS][0m Internet connectivity: PASSED
[0;34m[INFO][0m Testing: npm registry
[0;32m[SUCCESS][0m npm registry: PASSED
[0;34m[INFO][0m Testing: GitHub connectivity
[0;32m[SUCCESS][0m GitHub connectivity: PASSED

[0;35mTesting Chrome DevTools MCP[0m
[0;34m[INFO][0m Testing: Chrome DevTools MCP package
[0;32m[SUCCESS][0m Chrome DevTools MCP package: PASSED
[1;33m[WARNING][0m Chrome not found in standard locations

[0;35mTesting Playwright MCP[0m
[0;34m[INFO][0m Testing: Playwright MCP package
[0;32m[SUCCESS][0m Playwright MCP package: PASSED
[0;34m[INFO][0m Testing: Playwright package
[0;32m[SUCCESS][0m Playwright package: PASSED
[0;34m[INFO][0m Checking Playwright browser installations...
[0;32m[SUCCESS][0m Playwright CLI available

[0;35mTesting API Connectivity[0m
[1;33m[WARNING][0m Ahrefs API key not configured (AHREFS_API_KEY)
[1;33m[WARNING][0m Perplexity API key not configured (PERPLEXITY_API_KEY)
[1;33m[WARNING][0m Cloudflare API credentials not configured (CLOUDFLARE_API_TOKEN, CLOUDFLARE_ACCOUNT_ID)

[0;35mTesting MCP Configurations[0m
[0;32m[SUCCESS][0m MCP templates directory exists
[0;34m[INFO][0m Testing: JSON validation: ahrefs-seo.json
[0;32m[SUCCESS][0m JSON validation: ahrefs-seo.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: augment-context-engine.json
[0;32m[SUCCESS][0m JSON validation: augment-context-engine.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: chrome-devtools-advanced.json
[0;32m[SUCCESS][0m JSON validation: chrome-devtools-advanced.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: chrome-devtools.json
[0;32m[SUCCESS][0m JSON validation: chrome-devtools.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: claude-code-mcp.json
[0;32m[SUCCESS][0m JSON validation: claude-code-mcp.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: cloudflare-api.json
[0;32m[SUCCESS][0m JSON validation: cloudflare-api.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: cloudflare-browser-rendering.json
[0;32m[SUCCESS][0m JSON validation: cloudflare-browser-rendering.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: complete-mcp-config.json
[0;32m[SUCCESS][0m JSON validation: complete-mcp-config.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: crawl4ai-mcp-config.json
[0;32m[SUCCESS][0m JSON validation: crawl4ai-mcp-config.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: fluentcrm.json
[0;32m[SUCCESS][0m JSON validation: fluentcrm.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: google-search-console.json
[0;32m[SUCCESS][0m JSON validation: google-search-console.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: grep-vercel.json
[0;32m[SUCCESS][0m JSON validation: grep-vercel.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: ios-simulator-mcp.json
[0;32m[SUCCESS][0m JSON validation: ios-simulator-mcp.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: llm-tldr.json
[0;32m[SUCCESS][0m JSON validation: llm-tldr.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: mcporter.json
[0;32m[SUCCESS][0m JSON validation: mcporter.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: nextjs-devtools.json
[0;32m[SUCCESS][0m JSON validation: nextjs-devtools.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: openapi-search.json
[0;32m[SUCCESS][0m JSON validation: openapi-search.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: outscraper.json
[0;32m[SUCCESS][0m JSON validation: outscraper.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: pagespeed-mcp-config.json
[0;32m[SUCCESS][0m JSON validation: pagespeed-mcp-config.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: peekaboo.json
[0;32m[SUCCESS][0m JSON validation: peekaboo.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: perplexity-research.json
[0;32m[SUCCESS][0m JSON validation: perplexity-research.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: playwright-config.json
[0;32m[SUCCESS][0m JSON validation: playwright-config.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: playwright.json
[0;32m[SUCCESS][0m JSON validation: playwright.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: quickfile.json
[0;32m[SUCCESS][0m JSON validation: quickfile.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: shadcn.json
[0;32m[SUCCESS][0m JSON validation: shadcn.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: stagehand-both.json
[0;32m[SUCCESS][0m JSON validation: stagehand-both.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: stagehand-python.json
[0;32m[SUCCESS][0m JSON validation: stagehand-python.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: stagehand.json
[0;32m[SUCCESS][0m JSON validation: stagehand.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: twilio.json
[0;32m[SUCCESS][0m JSON validation: twilio.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: unstract.json
[0;32m[SUCCESS][0m JSON validation: unstract.json: PASSED
[0;34m[INFO][0m Testing: JSON validation: wordpress-mcp-adapter.json
[0;32m[SUCCESS][0m JSON validation: wordpress-mcp-adapter.json: PASSED

[0;35mValidation Report[0m

[0;34m[INFO][0m Total tests run: 39
[0;32m[SUCCESS][0m Tests passed: 39
[0;32m[SUCCESS][0m Tests failed: 0

[0;32m[SUCCESS][0m Overall status: EXCELLENT (100% success rate)

[0;32m[SUCCESS][0m All MCP integrations are ready to use!
[0;34m[INFO][0m Check .agents/MCP-INTEGRATIONS.md for usage examples
2. Observe JSON validation failure for chrome-devtools.json

## Expected Behavior

The validation script should correctly validate all MCP template JSON files.

## Environment

- aidevops version: 3.1.0
- OS: macOS

## Investigation

JSON validation via  passes. Issue may be:
1. Different file location being checked
2. Script running from wrong directory
3. Race condition or file permission issue

it was fixed locally, here is the output of the first run:

$ bash .agents/scripts/validate-mcp-integrations.sh 
MCP Integrations Validation

Testing Prerequisites
[INFO] Testing: Node.js availability
[SUCCESS] Node.js availability: PASSED
[INFO] Testing: npm availability
[SUCCESS] npm availability: PASSED
[INFO] Node.js version: v25.8.1
[INFO] npm version: 11.11.0

Testing Network Connectivity
[INFO] Testing: Internet connectivity
[SUCCESS] Internet connectivity: PASSED
[INFO] Testing: npm registry
[SUCCESS] npm registry: PASSED
[INFO] Testing: GitHub connectivity
[SUCCESS] GitHub connectivity: PASSED

Testing Chrome DevTools MCP
[INFO] Testing: Chrome DevTools MCP package
[SUCCESS] Chrome DevTools MCP package: PASSED
[WARNING] Chrome not found in standard locations

Testing Playwright MCP
[INFO] Testing: Playwright MCP package
[SUCCESS] Playwright MCP package: PASSED
[INFO] Testing: Playwright package
[SUCCESS] Playwright package: PASSED
[INFO] Checking Playwright browser installations...
[SUCCESS] Playwright CLI available

Testing API Connectivity
[WARNING] Ahrefs API key not configured (AHREFS_API_KEY)
[WARNING] Perplexity API key not configured (PERPLEXITY_API_KEY)
[WARNING] Cloudflare API credentials not configured (CLOUDFLARE_API_TOKEN, CLOUDFLARE_ACCOUNT_ID)

Testing MCP Configurations
[SUCCESS] MCP templates directory exists
[INFO] Testing: JSON validation: ahrefs-seo.json
[SUCCESS] JSON validation: ahrefs-seo.json: PASSED
[INFO] Testing: JSON validation: augment-context-engine.json
[SUCCESS] JSON validation: augment-context-engine.json: PASSED
[INFO] Testing: JSON validation: chrome-devtools-advanced.json
[SUCCESS] JSON validation: chrome-devtools-advanced.json: PASSED
[INFO] Testing: JSON validation: chrome-devtools.json
[ERROR] JSON validation: chrome-devtools.json: FAILED
$

--- a/.agents/scripts/validate-mcp-integrations.sh
+++ b/.agents/scripts/validate-mcp-integrations.sh
@@ -139,6 +139,23 @@ test_api_connectivity() {
 	return 0
 }
 
+# Validate a single JSON file directly (bypasses bash -c quoting issues)
+# Uses jq if available, falls back to python3 -c with proper argument passing
+validate_json_file() {
+	local file="$1"
+
+	if command -v jq &>/dev/null; then
+		jq empty "$file" 2>/dev/null
+		return $?
+	elif command -v python3 &>/dev/null; then
+		python3 -c "import json, sys; json.load(open(sys.argv[1]))" "$file" 2>/dev/null
+		return $?
+	else
+		# No JSON validator available — skip rather than false-fail
+		return 0
+	fi
+}
+
 # Test MCP configurations
 test_mcp_configurations() {
 	print_header "Testing MCP Configurations"
@@ -148,12 +165,25 @@ test_mcp_configurations() {
 	if [[ -d "$config_dir" ]]; then
 		print_success "MCP templates directory exists"
 
-		# Test each configuration file
+		# Check for JSON validator availability
+		if ! command -v jq &>/dev/null && ! command -v python3 &>/dev/null; then
+			print_warning "No JSON validator found (install jq or python3)"
+		fi
+
+		# Validate each configuration file directly (not via bash -c)
 		for config_file in "$config_dir"/*.json; do
 			if [[ -f "$config_file" ]]; then
 				local filename
 				filename=$(basename "$config_file")
-				run_test "JSON validation: $filename" "python3 -m json.tool '$config_file'"
+				((++total_tests))
+				print_info "Testing: JSON validation: $filename"
+				if validate_json_file "$config_file"; then
+					print_success "JSON validation: $filename: PASSED"
+					((++passed_tests))
+				else
+					print_error "JSON validation: $filename: FAILED"
+					((++failed_tests))
+				fi
 			fi
 		done
 	else

